src/dsp_tools/utils/xml_utils.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+import copy
+from pathlib import Path
+from typing import Any, Union
+
+from lxml import etree
+
+from dsp_tools.models.exceptions import UserError
+from dsp_tools.utils.create_logger import get_logger
+
+logger = get_logger(__name__)
+
+
+def parse_and_clean_xml_file(input_file: Union[str, Path, etree._ElementTree[Any]]) -> etree._Element:
+    """
+    Parse an XML file with DSP-conform data,
+    remove namespace URI from the elements' names,
+    and transform the special tags <annotation>, <region>, and <link>
+    to their technically correct form
+    <resource restype="Annotation">, <resource restype="Region">, and <resource restype="LinkObj">.
+
+    Args:
+        input_file: path to the XML file, or parsed ElementTree
+
+    Returns:
+        the root element of the parsed XML file
+    """
+
+    # remove comments and processing instructions (commented out properties break the XMLProperty constructor)
+
+    match input_file:
```

Comment:
@BalduinLandolt 
The pattern matching didn't work so far. When I run it isolated it is fine. But the e2e fail.
Do you see a quick fix? Otherwise I would suggest we table this and go back to the if else. Then we can close this PR from my point of view.

Reply:
I was unsure about the way you implemented the SPM, so I researched it, because I was curious if it's allowed to do it the way you do it. But it seems to be correct.

So no, I don't see why it fails. I suggest going back to the if-else, so that we can move forward.

Reply:
I think it will work, if you just remove the parenthesis after the type (you want to say "in case it is a string", etc. so you don't want to instantiate a ew string; `str` is the type string, `str()` is a new object of type string), it will work

Reply:
or even better:
```python
match input_file:
    case str(s):
        # here `s` is strictly typed as str, rather than as a union
        do_something(s)
```

Reply:
Unfortunately leaving the braces away is causing a syntax error. 
<img width="454" alt="Screenshot 2023-10-04 at 13 07 07" src="https://github.com/dasch-swiss/dsp-tools/assets/103038637/ecca38bd-8aea-41cd-b6a2-c23a312c5150">

Placing the s in the variable does not fix it unfortunately, I have a feeling it has something to do that we use a type from a package that is private. So I'll reverse it so that we can finish this quickly.